doc/developers/maintainer.rst

```diff
@@ -141,36 +150,49 @@ Making a release
    The wheel building workflow is managed by GitHub Actions and the results be browsed at:
    https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Wheel+builder%22
 
-   Once all works, you can proceed with tagging. Create the tag and push it (if
+.. note::
+
+  Before building the wheels, make sure that the ``pyproject.toml`` file is
+  up to date and using the oldest version of ``numpy`` for each Python version
+  to avoid ABI incompatibility issues. Moreover, a new line have to be included
+  in the ``pyproject.toml`` file for each new supported version of Python.
+
+4. Once the CD has completed successfully, upload the generated artifacts
```

Comment:
CD? Perhaps CI?
Or maybe you could explain what CD means? At least I don't know what it is... :(

Reply:
https://en.wikipedia.org/wiki/Continuous_delivery

I will rephrase this paragraph and the previous one.

Reply:
Sure @cmarmo, let me explain carefully.

CD means Continuous Delivery. It is the automated process by which the code is built (wheel), test, and then put into a staging or pre-production environment (Anaconda cloud).

Reply:
Should we put this information in the maintainers' guide or a link to a proper definition should be enough?